src/meta/src/hummock/manager/mod.rs

```diff
@@ -893,32 +936,196 @@ where
             .await
     }
 
+    // need mutex protect
+    async fn precheck_compaction_group(
+        &self,
+        compaction_group_id: CompactionGroupId,
+        compaction_statuses: &mut BTreeMap<CompactionGroupId, CompactStatus>,
+        compaction_config: &CompactionConfig,
+    ) -> Result<()> {
+        if !compaction_statuses.contains_key(&compaction_group_id) {
+            let mut compact_statuses = BTreeMapTransaction::new(compaction_statuses);
+            let new_compact_status = compact_statuses.new_entry_insert_txn(
+                compaction_group_id,
+                CompactStatus::new(compaction_group_id, compaction_config.max_level),
+            );
+            commit_multi_var!(self, None, Transaction::default(), new_compact_status)?;
+
+            // init compaction selector for new group
+        }
+
+        Ok(())
+    }
+
+    fn fetch_selector(
+        compaction_selectors: &mut HashMap<
+            CompactionGroupId,
+            HashMap<compact_task::TaskType, Box<dyn LevelSelector>>,
+        >,
+        compaction_group_id: CompactionGroupId,
+        task_type: compact_task::TaskType,
+        selector_option: SelectorOption,
+    ) -> &mut Box<dyn LevelSelector> {
+        match compaction_selectors
+            .entry(compaction_group_id)
+            .or_default()
+            .entry(task_type)
+        {
+            Occupied(mut selector) => selector.get_mut().try_update(selector_option),
```

Comment:
Do we need update `SelectorOption` every time?

Reply:
We just need to update on the selector when selector_option is updated, and the `try_update` logic will determine whether the selector_option is updated or not